pkg/frontend/admin_openshiftcluster_etcdcertificaterenew.go

```diff
@@ -0,0 +1,429 @@
+		}
+		if utilcert.IsCertExpired(certData[0]) {
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "secret %s is already expired, quitting.", secretname)
+		}
+	}
+	e.log.Infoln("Etcd certs exits, are not expired but close to expiry")
+
+	return nil
+}
+
+func (e *etcdrenew) fetchEtcdCurrentRevision(ctx context.Context) error {
+	e.log.Infoln("fetching etcd Current Revision now")
+	rawEtcd, err := e.k.KubeGet(ctx, "etcd.operator.openshift.io", "", "cluster")
+	if err != nil {
+		return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", err.Error())
+	}
+	etcd := &operatorv1.Etcd{}
+	err = codec.NewDecoderBytes(rawEtcd, &codec.JsonHandle{}).Decode(etcd)
+	if err != nil {
+		return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", fmt.Sprintf("failed to decode etcd object, %s", err.Error()))
+	}
+
+	e.lastRevision = etcd.Status.LatestAvailableRevision
+	e.log.Infof("Current Etcd Revision is %d", e.lastRevision)
+
+	return nil
+}
+
+// backup existing etcd secrets in the cluster, into runtime variable,
+func (e *etcdrenew) backupEtcdSecrets(ctx context.Context) error {
+	e.log.Infoln("backing up etcd secrets now")
+	for _, secretname := range e.secretNames {
+		return retry.OnError(wait.Backoff{
+			Steps:    10,
+			Duration: 2 * time.Second,
+		}, func(err error) bool {
+			return errors.IsBadRequest(err) || errors.IsInternalError(err) || errors.IsServerTimeout(err)
+		}, func() error {
+			e.log.Infof("Backing up secret %s", secretname)
+			data, err := e.k.KubeGet(ctx, "Secret", namespaceEtcds, secretname)
+			if err != nil {
+				return err
+			}
+			secret := &corev1.Secret{}
+			err = codec.NewDecoderBytes(data, &codec.JsonHandle{}).Decode(secret)
+			if err != nil {
+				return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", fmt.Sprintf("failed to decode secret, %s", err.Error()))
+			}
+			secret.CreationTimestamp = metav1.Time{
+				Time: time.Now(),
+			}
+			secret.ObjectMeta.ResourceVersion = ""
+			secret.ObjectMeta.UID = ""
+
+			var cert []byte
+			err = codec.NewEncoderBytes(&cert, &codec.JsonHandle{}).Encode(secret)
+			if err != nil {
+				return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", fmt.Sprintf("failed to encode secret, %s", err.Error()))
+			}
+			e.backupSecrets[secretname] = cert
+			return nil
+		})
+	}
+
+	e.log.Infoln("backing up etcd secrets done")
+	return nil
+}
+
+// delete the etcd secrets and on successful deletion,
+// valid secrets will be recreated and a new revision will be applied by the etcd operator
+func (e *etcdrenew) deleteEtcdSecrets(ctx context.Context) error {
+	e.log.Infoln("deleting etcd secrets now")
+	for _, secretname := range e.secretNames {
+		return retry.OnError(wait.Backoff{
```

Comment:
this shouldn't be a return as it only does one secret

Reply:
Done, made the change.